components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/confirmation/ResendConfirmationManager.java

```diff
@@ -147,6 +148,19 @@ public ResendConfirmationDTO resendConfirmation(String tenantDomain, String rese
         UserRecoveryData userRecoveryData = userAccountRecoveryManager
                 .getUserRecoveryData(resendCode, RecoverySteps.RESEND_CONFIRMATION_CODE);
         User user = userRecoveryData.getUser();
+        String recoveryFlowId = userRecoveryData.getRecoveryFlowId();
+        UserRecoveryFlowData userRecoveryFlowData = userAccountRecoveryManager.loadUserRecoveryFlowData(
+                userRecoveryData);
+        int resendCount = userRecoveryFlowData.getResendCount();
+        if (resendCount >= Integer.parseInt(Utils.getRecoveryConfigs(IdentityRecoveryConstants.ConnectorConfig.
```

Comment:
Since this is tenant level config, we need to pass the tenant domain as well. And since the config is used only once in the method, it'll be redundant to use a variable for now.